crates/evm/fuzz/src/strategies/mod.rs

```diff
@@ -8,14 +8,38 @@ mod param;
 pub use param::{fuzz_param, fuzz_param_from_state};
 
 mod calldata;
-pub use calldata::{
-    fuzz_calldata, fuzz_calldata_with_config, CalldataFuzzDictionary, CalldataFuzzDictionaryConfig,
-};
+pub use calldata::{fuzz_calldata, fuzz_calldata_from_state};
 
 mod state;
-pub use state::{
-    build_initial_state, collect_created_contracts, fuzz_calldata_from_state, EvmFuzzState,
-};
+pub use state::{build_initial_state, collect_created_contracts, EvmFuzzState};
 
 mod invariants;
 pub use invariants::{fuzz_contract_with_calldata, invariant_strat, override_call_strat};
+
+/// Macro to create strategy with fixtures.
+/// 1. A default strategy if no fixture defined for current parameter.
+/// 2. A weighted strategy that use fixtures and default strategy values for current parameter.
+/// If fixture is not of the same type as fuzzed parameter then value is rejected and error raised.
+macro_rules! fixture_strategy {
+    ($fixtures:ident, $strategy_value:expr, $default_strategy:expr) => {
+        if let Some(fixtures) = $fixtures {
+            proptest::prop_oneof![
+                50 => {
+                    let custom_fixtures: Vec<DynSolValue> =
+                        fixtures.iter().enumerate().map(|(_, value)| value.to_owned()).collect();
+                    let custom_fixtures_len = custom_fixtures.len();
+                    any::<prop::sample::Index>()
+                        .prop_filter_map("invalid fixture", move |index| {
+                            let index = index.index(custom_fixtures_len);
+                            $strategy_value(custom_fixtures.get(index))
+                    })
+                },
+                50 => $default_strategy
+            ].boxed()
```

Comment:
shouldn't we apply `prop_filter_map` to the entier `prop_oneof!` output? I believe right now if we only have invalid fixtures it will fail with max rejects while it'd be better if it used default strategy instead

Reply:
was thinking that could be better to fail (faster) with max rejects rather than run tests with a config that is not really desired, but can change to apply to both strategies, wdyt?

Reply:
right, let's keep it like this for now